gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigResourceLocalHandler.java

```diff
@@ -187,6 +187,9 @@ public static FlowSpec createFlowSpecForConfig(FlowConfig flowConfig) {
       Schedule schedule = flowConfig.getSchedule();
       configBuilder.addPrimitive(ConfigurationKeys.JOB_SCHEDULE_KEY, schedule.getCronSchedule());
       configBuilder.addPrimitive(ConfigurationKeys.FLOW_RUN_IMMEDIATELY, schedule.isRunImmediately());
+      if (isRunOnceFlow(flowConfig)) {
```

Comment:
Please add a comment here explaining why an id is assigned for this case.

Reply:
done